makefiles/Makefile.devhub-content-new

```diff
@@ -0,0 +1,59 @@
+GIT_BRANCH=$(shell git rev-parse --abbrev-ref HEAD)
+USER=$(shell whoami)
+
+STAGING_URL="http://docs-devhub-staging.s3-website-us-east-1.amazonaws.com"
+STAGING_BUCKET=docs-devhub-staging
+
+STRAPI_URL=http://54.219.137.111:1337
+
+PRODUCTION_BUCKET=developer-hub-prod
+PRODUCTION_URL=https://mongodb.com/developer
+
+PROJECT=devhub
+MUT_PREFIX ?= $(PROJECT)
+REPO_DIR=$(shell pwd)
+
+COMMIT_HASH=$(shell git rev-parse --short HEAD)
+
+GITHUB_USER = $(shell printenv GITHUB_BOT_USERNAME)
+GITHUB_PASS = $(shell printenv GITHUB_BOT_PASSWORD)
+SNOOTY_DB_USR = $(shell printenv MONGO_ATLAS_USERNAME)
+SNOOTY_DB_PWD = $(shell printenv MONGO_ATLAS_PASSWORD)
+
+include ~/shared.mk
+
+update-snooty:
+	cd ../../snooty-devhub; \
+	git fetch --all; \
+	git reset --hard origin/master; \
+	npm install --production
+
+get-build-dependencies:
+	@curl https://raw.githubusercontent.com/mongodb/docs-worker-pool/meta/publishedbranches/devhub-content.yaml > ${REPO_DIR}/published-branches.yaml
+
+next-gen-html: update-snooty
+	# snooty parse and then build-front-end
+	@echo ${SNOOTY_DB_PWD} | snooty build "${REPO_DIR}" "mongodb+srv://${SNOOTY_DB_USR}:@cluster0-ylwlz.mongodb.net/snooty?retryWrites=true" --commit "${COMMIT_HASH}"; \
+	if [ $$? -eq 1 ]; then \
+		exit 1; \
+	else \
+		exit 0; \
+	fi
+	rsync -az --exclude '.git' ${REPO_DIR}/../../snooty-devhub/ ${REPO_DIR}/snooty-devhub-tmp;
+	cp ${REPO_DIR}/.env.production ${REPO_DIR}/snooty-devhub-tmp; 
+	curl https://raw.githubusercontent.com/mongodb/devhub/master/gatsby-config.prod-new.js > ${REPO_DIR}/snooty-devhub-tmp/gatsby-config.js
+	cd snooty-devhub-tmp; \
+	echo "GATSBY_SITE=${PROJECT}" >> .env.production; \
+	echo "COMMIT_HASH=${COMMIT_HASH}" >> .env.production; \
+	echo "STRAPI_URL=${STRAPI_URL}" >> .env.production; \
+	npm run build; \
+	cp -r "${REPO_DIR}/snooty-devhub-tmp/public" ${REPO_DIR};
+  
+next-gen-stage: ## Host online for review
+	mut-publish public ${STAGING_BUCKET} --prefix="${COMMIT_HASH}/${MUT_PREFIX}" --stage ${ARGS};
+	echo "${COMMIT_HASH}/${MUT_PREFIX}"
+	echo "Hosted at ${STAGING_URL}/${COMMIT_HASH}/${MUT_PREFIX}/${USER}/${GIT_BRANCH}/";
+
+next-gen-deploy:	
+	yes | AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID} AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY} mut-publish public ${PRODUCTION_BUCKET} --prefix=/developer --deploy --deployed-url-prefix=https://mongodb.com/developer --json --all-subdirectories ${ARGS};
```

Comment:
Hey team, after speaking with IT Systems Eng, they recommended we do something like this where we parameterize the AWS credentials we are using for the job and pass in two new ones so we properly upload as the AWS user. This is also important since docs is going to consolidate infrastructure soon and the user associated with buckets may change.

Of course I can hand these over via 0bin when it is time to deploy but wanted to get a check on this just to be sure this is the way to do it here (unless we can set ENV secrets before running this portion of the job)

Reply:
I think this is a good idea, but let's scope the parameters as well to something like `AWS_SECRET_ACCESS_KEY_DEVHUB` and `AWS_ACCESS_KEY_ID_DEVHUB` 

Just to be hyper specific and avoid any scope collisions, or a regression from an inadvertent change to the values in a config in the future when the infra consolidation occurs.

Reply:
Great idea, just added it thanks

Reply:
Looks good